src/Discord.Net.Core/Entities/Messages/EmbedBuilder.cs

```diff
@@ -150,7 +150,7 @@ public int Length
                 int authorLength = Author?.Name?.Length ?? 0;
                 int descriptionLength = Description?.Length ?? 0;
                 int footerLength = Footer?.Text?.Length ?? 0;
-                int fieldSum = Fields.Sum(f => f.Name.Length + f.Value.ToString().Length);
+                int fieldSum = Fields.Sum(f => f.Name.Length + (f.Value == null ? 0 : f.Value.ToString()?.Length ?? 0));
```

Comment:
the initial null check seems redundant. `f.Value?.ToString().Length ?? 0` should be enough. Can you please verify whether that works or not and update your code if so.